units/sdlaudio.inc

```diff
@@ -549,20 +1307,153 @@ procedure SDL_ClearQueuedAudio(dev: TSDL_AudioDeviceID); cdecl;
+ *
+ * New programs might want to use SDL_UnlockAudioDevice() instead. This
+ * function is equivalent to calling...
+ *
+ * ```c
+ * SDL_UnlockAudioDevice(1);
+ * ```
+ *
+ * ...and is only useful if you used the legacy SDL_OpenAudio() function.
+ *
+ * \since This function is available since SDL 2.0.0.
+ *
+ * \sa SDL_LockAudio
+ * \sa SDL_UnlockAudioDevice
+ *}
+procedure SDL_UnlockAudio; cdecl;
+  external SDL_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_SDL_Unlock' {$ENDIF} {$ENDIF};
+
+{**
+ * Use this function to unlock the audio callback function for a specified
+ * device.
+ *
+ * This function should be paired with a previous SDL_LockAudioDevice() call.
+ *
+ * \param dev the ID of the device to be unlocked
+ *
+ * \since This function is available since SDL 2.0.0.
+ *
+ * \sa SDL_LockAudioDevice
+ *}
+procedure SDL_UnlockAudioDevice(dev: TSDL_AudioDeviceID); cdecl;
+  external SDL_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_SDL_UnlockAudioDevice' {$ENDIF} {$ENDIF};
+
+{*Audio lock functions*}
+
+{**
+ * This function is a legacy means of closing the audio device.
+ *
+ * This function is equivalent to calling...
+ *
+ * ```c
+ * SDL_CloseAudioDevice(1);
+ * ```
+ *
+ * ...and is only useful if you used the legacy SDL_OpenAudio() function.
+ *
+ * \since This function is available since SDL 2.0.0.
+ *
+ * \sa SDL_OpenAudio
+ *}
+procedure SDL_CloseAudio; cdecl;
+  external SDL_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_SDL_CloseAudio' {$ENDIF} {$ENDIF};
+
+{**
+ * Use this function to shut down audio processing and close the audio device.
+ *
+ * The application should close open audio devices once they are no longer
+ * needed. Calling this function will wait until the device's audio callback
+ * is not running, release the audio hardware and then clean up internal
+ * state. No further audio will play from this device once this function
+ * returns.
+ *
+ * This function may block briefly while pending audio data is played by the
+ * hardware, so that applications don't drop the last buffer of data they
+ * supplied.
+ *
+ * The device ID is invalid as soon as the device is closed, and is eligible
+ * for reuse in a new SDL_OpenAudioDevice() call immediately.
+ *
+ * \param dev an audio device previously opened with SDL_OpenAudioDevice()
+ *
+ * \since This function is available since SDL 2.0.0.
+ *
+ * \sa SDL_OpenAudioDevice
+ *}
+procedure SDL_CloseAudioDevice(dev: TSDL_AudioDeviceID); cdecl;
+  external SDL_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_SDL_CloseAudioDevice' {$ENDIF} {$ENDIF};
 
   {**
    *  1 if audio device is still functioning, zero if not, -1 on error.
-   *}
-function SDL_AudioDeviceConnected(dev: TSDL_AudioDeviceID): cint cdecl; external SDL_LibName {$IFDEF DELPHI} {$IFDEF MACOS} name '_SDL_AudioDeviceConnected' {$ENDIF} {$ENDIF};
+   *
+
+   SDL2-For-Pascal: Function is gone, obviously deprecated.
+   }
+function SDL_AudioDeviceConnected(dev: TSDL_AudioDeviceID): cint; cdecl;
```

Comment:
Was this ever actually a thing? Running `git grep` on the C library sources returns nothing.
```
git grep 'AudioDeviceConnected' -- include/SDL_audio.h
```

Reply:
I wonder, your grep won't look in the history, right? - I marked it `deprecated`, since I assumed `AudioDeviceConnected` being present in the past. 

Although, on closer examination I couldn't find any evidence for `AudioDeviceConnected` ever being present in this file and removing it from sdlaudio.inc is possible, - It has been present in SDL_sysaudio.h in the source directory once.

Still I wonder, how your grep led you to this conclusion.

Best regards
Matthias

Reply:
On one hand - you're right, the proper command would be:
```
git grep -e 'AudioDeviceConnected' $(git rev-list --all) -- include/SDL_audio.h
```
On the other hand, this also produces zero results.

Looking in `src/audio/` produces a match in `sdl/audio/SDL_sysaudio.h`, but then, that makes it a part of SDL's internal API and not exposed to library users.

Reply:
Alright, thanks :-)

Reply:
Removed in 0ea354db13383fdeda9c34c68df3dd81b888ae3c.